engine/src/sellst.cpp

```diff
@@ -133,7 +133,12 @@ void MCSellist::Clean()
             MCSelnode* t_next = t_cursor->next();
             t_cursor->remove(objects);
             delete t_cursor;
-            t_cursor = t_next;
+            if (objects == nil)
+            	t_cursor = nil;
+			else if (objects == t_next)
+				t_cursor = objects->next();
```

Comment:
Why do we have to loop round again in this situation?

Reply:
I may have to rethink this - we need to detect when the first object has been removed, so have to check if `objects` is now null (or now points to the next object) but make sure the loop still continues if there is now just a single object remaining (which hasn't been checked yet)
Basically, iterating over a double linked list while removing items from it is a pain :(